File: queue-reconstruction-by-height/queue-reconstruction-by-height.py
class Solution:
    def reconstructQueue(self, people: List[List[int]]) -> List[List[int]]:
        h = []
        for p in people:
            heapq.heappush(h, [-p[0], p[1]])
        res = []
        while h:
            p = heapq.heappop(h)
            res.insert(p[1], [-p[0], p[1]])
        return res
        
        

File: queue-reconstruction-by-height/queue-reconstruction-by-height.java
class Solution {
    public int[][] reconstructQueue(int[][] people) {
        PriorityQueue<int[]> q = new PriorityQueue<>(
            (a, b) -> a[0] != b[0] ? -a[0]+b[0]:a[1]-b[1]);
        List<int[]> li = new LinkedList<>();
        
        for (int[] p: people) q.offer(p);
        
        while (!q.isEmpty()) {
            int[] a = q.poll();
            li.add(a[1], a);
        }
        
        return li.toArray(new int[people.length][]);
    }
}